src/commands/StepCommand.java

```diff
@@ -80,11 +80,11 @@ public void execute(String[] input, SimulationManager simulationManager) {
             return;
         }
 
-        if (simulationManager.isTrace()) {
-            System.out.println(simulationManager.getCurrentStateDiffString());
-            return;
-        }
+        if (simulationManager.isDiffMode()) {
+            System.out.println(simulationManager.getCurrentStateDiffStringByDelta(steps));
```

Comment:
Is there any case where `getCurrentStateDiffStringFromLastCommit` wouldn't work here?

I'm wondering if `getCurrentStateDiffStringByDelta` is necessary.

Reply:
Yup, `getCurrentStateDiffStringFromLastCommit` doesn't work when stepping through traces since we don't generate and commit nodes as we step through a trace. I've added a comment for clarity.